lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1315,17 +1315,19 @@ private function getSQLForJoins(string $fromAlias, array &$knownAliases) : strin
 
         if (isset($this->sqlParts['join'][$fromAlias])) {
             foreach ($this->sqlParts['join'][$fromAlias] as $join) {
-                if (array_key_exists($join['joinAlias'], $knownAliases)) {
-                    throw NonUniqueAlias::new($join['joinAlias'], array_keys($knownAliases));
+                /** @var QueryPartJoin $join */
+                if (array_key_exists($join->joinAlias, $knownAliases)) {
+                    throw NonUniqueAlias::new($join->joinAlias, array_keys($knownAliases));
                 }
-                $sql                             .= ' ' . strtoupper($join['joinType'])
-                    . ' JOIN ' . $join['joinTable'] . ' ' . $join['joinAlias']
-                    . ' ON ' . ((string) $join['joinCondition']);
-                $knownAliases[$join['joinAlias']] = true;
+                $sql                           .= ' ' . strtoupper($join->joinType)
```

Comment:
The upper-case value of the join type could be coded in a named constructor instead of the builder.

Reply:
Done in f1078a65e4e65064021a59b613b0b441c499ad11 and 545481915591df7300339c1665b045ce5a38c5e8.